web/server/entrypoint_test.go

```diff
@@ -201,122 +201,169 @@ func isExpectedShutdownError(err error, testLogger logging.Logger) bool {
+					NetworkConfigData: config.NetworkConfigData{
+						BindAddress: machineAddress,
+					},
+				},
+			}
 
-	rc := robottestutils.NewRobotClient(t, logger, machineAddress, time.Second)
+			cfgBytes, err := json.Marshal(&cfg)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, os.WriteFile(tempConfigFile.Name(), cfgBytes, 0o755), test.ShouldBeNil)
 
-	// Assert that, from client's perspective, robot is in an initializing state until
-	// `slowpoke` completes construction.
-	machineStatus, err := rc.MachineStatus(ctx)
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, machineStatus, test.ShouldNotBeNil)
-	test.That(t, machineStatus.State, test.ShouldEqual, robot.StateInitializing)
+			args := []string{"viam-server", "-config", tempConfigFile.Name()}
+			test.That(t, server.RunServer(ctx, args, logger), test.ShouldBeNil)
+		}()
 
-	// Assert that the `foo` package file exists during initialization, machine assumes
-	// package files may still be in use.)
-	_, err = os.Stat(fakeModuleDataFile.Name())
-	test.That(t, err, test.ShouldBeNil)
+		// Set `DoNotWaitForRunning` to true to allow connecting to a still-initializing
+		// machine.
+		client.DoNotWaitForRunning.Store(true)
+		defer func() {
+			client.DoNotWaitForRunning.Store(false)
+		}()
 
-	// Allow `slowpoke` to complete construction.
-	close(completeConstruction)
+		rc := robottestutils.NewRobotClient(t, logger, machineAddress, time.Second)
 
-	gtestutils.WaitForAssertion(t, func(tb testing.TB) {
+		// Assert that, from client's perspective, robot is in an initializing state until
+		// `slowpoke` completes construction.
 		machineStatus, err := rc.MachineStatus(ctx)
-		test.That(tb, err, test.ShouldBeNil)
-		test.That(tb, machineStatus, test.ShouldNotBeNil)
-		test.That(tb, machineStatus.State, test.ShouldEqual, robot.StateRunning)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, machineStatus, test.ShouldNotBeNil)
+		test.That(t, machineStatus.State, test.ShouldEqual, robot.StateInitializing)
+
+		// Assert that the `foo` package file exists during initialization, machine assumes
+		// package files may still be in use.)
+		_, err = os.Stat(fakeModuleDataFile.Name())
+		test.That(t, err, test.ShouldBeNil)
+
+		// Allow `slowpoke` to complete construction.
+		close(completeConstruction)
+
+		gtestutils.WaitForAssertion(t, func(tb testing.TB) {
+			machineStatus, err := rc.MachineStatus(ctx)
+			test.That(tb, err, test.ShouldBeNil)
+			test.That(tb, machineStatus, test.ShouldNotBeNil)
+			test.That(tb, machineStatus.State, test.ShouldEqual, robot.StateRunning)
+		})
+
+		// Assert that the `foo` file was removed, as the non-initializing `Reconfigure`
+		// determined it was unnecessary (no associated package/module.)
+		_, err = os.Stat(fakeModuleDataFile.Name())
+		test.That(t, os.IsNotExist(err), test.ShouldBeTrue)
+
+		// Cancel context and wait for server goroutine to stop running.
+		cancel()
+		wg.Wait()
 	})
+	t.Run("no resources", func(t *testing.T) {
```

Comment:
This diff is horrid to read; you can try using the side-by-side view, and I'll point out that I'm only _adding_ the "no resources" test here (the original test got moved to a subtest called "slow building resource".)

Reply:
You can add ?w=1 to the URL (which is a `git diff` option: `-w`) for ignoring whitespace changes:

https://github.com/viamrobotics/rdk/pull/4834/files?w=1

Reply:
More randomly, given there's no common setup between the tests, it seems a bit simpler to just have created a separate test function.

Reply:
Good to know about the URL, and I split the test into two 🫡 .